example/src/main/java/io/netty/example/stomp/websocket/StompChatHandler.java

```diff
@@ -0,0 +1,193 @@
+        StompVersion handshakeAcceptVersion = ctx.channel().attr(StompVersion.CHANNEL_ATTRIBUTE_KEY).get();
+        if (acceptVersions == null || !acceptVersions.contains(handshakeAcceptVersion.version())) {
+            sendErrorFrame("invalid version",
+                           "Received invalid version, expected " + handshakeAcceptVersion.version(), ctx);
+            return;
+        }
+
+        StompFrame connectedFrame = new DefaultStompFrame(StompCommand.CONNECTED);
+        connectedFrame.headers()
+                      .set(VERSION, handshakeAcceptVersion.version())
+                      .set(SERVER, "Netty-Server")
+                      .set(HEART_BEAT, "0,0");
+        ctx.writeAndFlush(connectedFrame);
+    }
+
+    private void onSubscribe(StompFrame inboundFrame, ChannelHandlerContext ctx) {
+        String destination = inboundFrame.headers().getAsString(DESTINATION);
+        String subscriptionId = inboundFrame.headers().getAsString(ID);
+
+        if (destination == null || subscriptionId == null) {
+            sendErrorFrame("missed header", "Required 'destination' or 'id' header missed", ctx);
+            return;
+        }
+
+        Set<StompSubscription> subscriptions =
+                CHAT_DESTINATIONS.putIfAbsent(destination, new HashSet<StompSubscription>());
+        if (subscriptions == null) {
+            subscriptions = CHAT_DESTINATIONS.get(destination);
+        }
+
+        final StompSubscription subscription = new StompSubscription(subscriptionId, destination, ctx.channel());
+        if (subscriptions.contains(subscription)) {
+            sendErrorFrame("duplicate subscription",
+                           "Received duplicate subscription id=" + subscriptionId, ctx);
+            return;
+        }
+
+        subscriptions.add(subscription);
+        ctx.channel().closeFuture().addListener(new ChannelFutureListener() {
+            @Override
+            public void operationComplete(ChannelFuture future) {
+                CHAT_DESTINATIONS.get(subscription.destination()).remove(subscription);
+            }
+        });
+
+        String receiptId = inboundFrame.headers().getAsString(RECEIPT);
+        if (receiptId != null) {
+            StompFrame receiptFrame = new DefaultStompFrame(StompCommand.RECEIPT);
+            receiptFrame.headers().set(RECEIPT_ID, receiptId);
+            ctx.writeAndFlush(receiptFrame);
+        }
+    }
+
+    private void onSend(StompFrame inboundFrame, ChannelHandlerContext ctx) {
+        String destination = inboundFrame.headers().getAsString(DESTINATION);
+        if (destination == null) {
+            sendErrorFrame("missed header", "required 'destination' header missed", ctx);
+            return;
+        }
+
+        Set<StompSubscription> subscriptions = CHAT_DESTINATIONS.get(destination);
+        for (StompSubscription subscription : subscriptions) {
+            subscription.channel().writeAndFlush(transformToMessage(inboundFrame, subscription));
```

Comment:
imho you can just call `write(...)` and only flush at the end once you iterated over all of these.

Reply:
`ctx.flush()`, Does it mean flush for all channels ? I need broadcast to each channel and after it do flush for each or I'm wrong ?

Reply:
You are right... sorry I missed that there a different Channels.